base_tier_validation/models/tier_validation.py

```diff
@@ -42,19 +42,14 @@ class TierValidation(models.AbstractModel):
         domain=lambda self: [("model", "=", self._name)],
         auto_join=True,
     )
-    to_validate_message = fields.Html(compute="_compute_validated_rejected")
-    # TODO: Delete in v17 in favor of validation_status field
+    to_validate_message = fields.Html(compute="_compute_to_validate_message")
     validated = fields.Boolean(
-        compute="_compute_validated_rejected", search="_search_validated"
+        compute="_compute_validated", store=True, compute_sudo=True
     )
-    validated_message = fields.Html(compute="_compute_validated_rejected")
+    validated_message = fields.Html(compute="_compute_validated_message")
     need_validation = fields.Boolean(compute="_compute_need_validation")
-    # TODO: Delete in v17 in favor of validation_status field
-    rejected = fields.Boolean(
```

Comment:
This TODO has been removed, do you think is not feasible to adapt everything to work with the validation_status and completely removing these booleans?

Not saying that it needs to be done right now in this PR, it is more like "is it a bad roadmap for any reason?"

Reply:
IMO the TODO did not refer to remove the field, it referred to use the validation_status, now the field is compute+store and has the appropriate depends, so I think it is not necessary to do anything else.

In addition, I think it is appropriate to have these fields, they are used in part of the code and are still necessary, and also, it is useful that they exist stored to be able to filter by that data if for example, you have a list of reviews.

Reply:
@LoisRForgeFlow As I pondered in another (resolved) comment, if anything, `_compute_rejected` and `_compute_validated` serve as entry points for customizations of the flow, replacing `_calc_reviews_validated` and `_calc_reviews_rejected` that are removed in this change. So I agree with @victoralmau.

Reply:
But then `validation_status` may have `rejected` but `rejected` to be False? I see that as an inconsistency. If not removing this in this version for compatibility reasons, it has to be removed ASAP.

Reply:
Exactly, if you look how the validation_status is computed, it could be computed based on the reviews status and forget about a rejected, approved, need review booleans. It is redundant. Of course the refactor is not easy, because it will affect views and a lot of places in the code, but we can have same functionality and extension points with only `validation_status`.

Reply:
If you want to go that way, that was the approach of https://github.com/OCA/server-ux/pull/1101, and it was not overly complicated.

Reply:
But .... same as any compute store field, right? Maybe I'm missing something, but I still don't see the problem with keeping these fields and making the approach of this PR more “_simple_” than removing them and changing everything necessary in the views. Also, what happens if another module adds a validation state x and in that new state it is rejected? With compute/write fields it could be done, otherwise, it would be more complicated/incorrect (e.g. https://github.com/OCA/server-ux/pull/1101/files#diff-af78280ddd15c60f3e3b3cfc9ffb3808a0bffac45e723028cadb63e3c689e41cR32).

Reply:
@pedrobaeza 
> But then validation_status may have rejected but rejected to be False? I see that as an inconsistency. If not removing this in this version for compatibility reasons, it has to be removed ASAP.

This is not actually the case. if one were to override `_compute_rejected`, the resulting value may be `False` while an individual tier review was in fact rejected. Because `validation_status` is based on the values of `approved` and `rejected`, this will prevent `validation_status` from being updated to `rejected`. A use case may be to allow one rejection to be offset by multliple approvals.

Reply:
I have checked again about this, and I think we have to remove the `rejected` and `validated` fields, and more now that you want to make them stored. You can keep `calc_reviews_validated` and `_calc_reviews_rejected` method the same for inheritance (and compatibility), but this time checking `validation_status` instead.

Reply:
Okay, those changes have been applied:

- The `rejected` and `validated` fields have been removed
- The `calc_reviews_validated` and `_calc_reviews_rejected` methods remain unchanged